src/OpenLoco/Ui/TextInput.cpp

```diff
@@ -166,4 +166,25 @@ namespace OpenLoco::Ui::TextInput
                 }),
             buffer.end());
     }
+
+    uint16_t InputSession::calculateInputLenLimit(string_id callerTitle, string_id field)
+    {
+        if (field == StringIds::buffer_2039)
+        {
+            auto& mOptions = S5::getOptions();
+
+            if (callerTitle == StringIds::scenario_details_title)
+            {
+                return uint16_t(std::size(mOptions.scenarioDetails) - 1);    
+            }
+
+            if (callerTitle == StringIds::scenario_name_title)
+            {
+                return uint16_t(std::size(mOptions.scenarioName) - 1);
+            }
```

Comment:
I really think this should be up to the caller to _set_ (e.g. a window function that initiates a text input session). The InputSession shouldn't have to _infer_ it based on a string.

Reply:
Would it be bad if the caller determine length by stringsIds or should I find different solution?

Reply:
You can assume the caller has access to this information. Consider, for example:
https://github.com/OpenLoco/OpenLoco/blob/71813b28e73f54451a6ce9311cccdde0ff186167/src/OpenLoco/Windows/CompanyWindow.cpp#L2626-L2630

You could change this to include another parameter for input size:
```cpp
        static void renameCompanyPrompt(Window* self, WidgetIndex_t widgetIndex)
        {
            auto company = CompanyManager::get(CompanyId(self->number));
            auto inputSize = StringManager::kUserStringSize;
            TextInput::openTextInput(self, StringIds::title_name_company,
                StringIds::prompt_enter_new_company_name, company->name,
                //  /------------ new parameter
                // V
                inputSize, widgetIndex, nullptr);
        }
```